.gitignore

```diff
@@ -71,9 +71,32 @@ logs
 # For a more precise, explicit template, see:
 # https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
 
-## General
-.idea/*
-.idea_modules/*
+
+## User Specific
+.idea/**/dictionaries
+.idea/**/workspace.xml
+.idea/**/usage.statistics.xml
+.idea/**/tasks.xml
+.idea/**/shelf
+
+
+# Gradle
+.idea/**/gradle.xml
+.idea/**/libraries
+
+
+## Uncomment if using auto-import.
+# .idea/py_ecc.iml
```

Comment:
This line looks specific to one project, which seems wrong for going into a template.

Also, I take it that you do actually want to commit some files in `.idea/` and that's why the nitpicking ignore file here? It makes me sad that it adds so much clutter :/ .  Also, I'm not sure we want new contributors accidentally adding (any of) their .idea files by default across all projects based on this template.

I'll think about it a bit.

Reply:
Oops, thanks for catching that. I've generalized the file name as` idea/*.iml`. I also uncommented this section, instead saying "Comment out if not using auto-import" to prevent new contributors from accidentally adding their `idea/*.im`l files. It is still a bit of clutter. I can remove the new comment from the file if you think it isn't necessary.